tests/sitemaps/test-class-wpseo-sitemaps-cache.php

```diff
@@ -0,0 +1,246 @@
+<?php
+
+/**
+ * apply_filters( 'wpseo_enable_xml_sitemap_transient_caching', true ); -- only applicable for retrieval (awesome!)
+ *
+ * Public
+ *
+ * ::register_clear_on_option_update .done
+ * ::clear_transient_cache - can't be done.
+ * ::get_validator_key .done
+ * ::get_storage_key .done
+ * ::clear .done
+ *
+ * Private (determine by coverage)
+ *
+ * ::invalidate_sitemap_cache .implied
+ * ::new_sitemap_cache_validator .implied
+ * ::get_sitemap_cache_validator .implied
+ * ::get_safe_sitemap_cache_type .implied
+ */
+class WPSEO_Sitemaps_Cache_Test extends WPSEO_UnitTestCase {
+
+	/**
+	 * Clean up
+	 */
+	public function tearDown() {
+		parent::tearDown();
+
+		remove_action( 'update_option', array( 'WPSEO_Sitemaps_Cache', 'clear_on_option_update' ) );
+	}
+
+	/**
+	 * Test the building of cache keys
+	 */
+	public function test_get_validator_key_global() {
+		$result = WPSEO_Sitemaps_Cache::get_validator_key();
+
+		$this->assertEquals( 'wpseo_sitemap_cache_validator_global', $result );
+	}
+
+	/**
+	 * Test the building of cache keys
+	 */
+	public function test_get_validator_key_type() {
+		$type     = 'blabla';
+		$expected = sprintf( 'wpseo_sitemap_%s_cache_validator', $type );
+
+		$result = WPSEO_Sitemaps_Cache::get_validator_key( $type );
+
+		$this->assertEquals( $expected, $result );
+	}
+
+	/**
+	 * Normal cache key retrieval
+	 */
+	public function test_get_storage_key() {
+		$n                = 1;
+		$type             = 'page';
+		$global_validator = 'global';
+		$type_validator   = 'type';
+
+		$global_validator_key = WPSEO_Sitemaps_Cache::get_validator_key();
+		update_option( $global_validator_key, $global_validator );
+
+		$type_validator_key = WPSEO_Sitemaps_Cache::get_validator_key( $type );
+		update_option( $type_validator_key, $type_validator );
+
+		$prefix  = WPSEO_Sitemaps_Cache::get_storage_key_prefix();
+		$postfix = sprintf( '_%d:%s_%s', $n, $global_validator, $type_validator );
+
+		$expected = $prefix . $type . $postfix;
+
+		// Act.
+		$result = WPSEO_Sitemaps_Cache::get_storage_key( $type, $n );
+
+		// Assert.
+		$this->assertEquals( $expected, $result );
+	}
+
+	/**
+	 * Key length should never be over 45 characters
+	 *
+	 * This would be 53 if we don't use a timeout, but we can't because all sitemaps would be autoloaded every request.
+	 */
+	public function test_get_storage_key_very_long_type() {
+		$n    = 1;
+		$type = str_repeat( 'a', 60 );
+
+		// Act.
+		$result = WPSEO_Sitemaps_Cache::get_storage_key( $type, $n );
+
+		// Assert.
+		$this->assertEquals( 45, strlen( $result ) );
```

Comment:
This can be much more specific, you can actually use `abcdefghijklmnopqrstabcdefghijklmnopqrstabcdefghijklmnopqrst` as a type and check that the result matches the expected result.